app/src/main/java/com/example/sharingang/ItemsListFragment.kt

```diff
@@ -42,8 +43,20 @@ class ItemsListFragment : Fragment() {
             goToMap(view)
         }
         binding.userProfileButton.setOnClickListener {
+            val sharedPreferences = activity?.getSharedPreferences(
+                getString(R.string.preference_user_info),
+                Context.MODE_PRIVATE
+            )
             val action =
+                if (sharedPreferences == null) {
                     ItemsListFragmentDirections.actionItemsListFragmentToUserProfileFragment("test")
+                } else {
+                    ItemsListFragmentDirections.actionItemsListFragmentToUserProfileFragment(
+                        sharedPreferences.getString(
+                            getString(R.string.account_firebase_uid), "test"
```

Comment:
If the user is not logged in, I think we shouldn't go to the user profile at all, maybe display a message with a Toast or Snackbar.
Also for the tests, I'll try to see if we can use dependency injection to get a fake user ID for the tests.